php_phongo.c

```diff
@@ -665,9 +665,11 @@ bool phongo_execute_bulk_write(mongoc_client_t* client, const char* namespace, p
 		if ((error.domain == MONGOC_ERROR_COMMAND && error.code != MONGOC_ERROR_COMMAND_INVALID_ARG) ||
 			error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN) {
 			phongo_throw_exception(PHONGO_ERROR_WRITE_FAILED TSRMLS_CC, "%s", error.message);
+			phongo_add_exception_prop(ZEND_STRL("resultDocument"), return_value TSRMLS_CC);
```

Comment:
I don't think this is quite right, as we are already adding the ``writeResult`` to this exception in the next line (669). The issue is a bit sparse, and talks about a new getter instead of a property too. In any case, I would have expected that line 672 would use ``writeResult`` instead of ``resultDocument``.

Reply:
`phongo_execute_command()` basically needs a copy of this `if (!success)` conditional.

I'd start by only using a CommandException for `error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN`. Otherwise, we can fall through to `phongo_throw_exception_from_bson_error_t()`, as this area of the code does.

This means if  `phongo_execute_command()` is throwing an instance of ServerException, it's actually a CommandException.